context/app/static/js/components/detailPage/visualization/Visualization/Visualization.jsx

```diff
@@ -82,27 +67,33 @@ function Visualization({ vitData, uuid, hasNotebook, shouldDisplayHeader, should
     }
   }, [hasNotebook, vizIsFullscreen, setVizNotebookId, uuid]);
 
-  const [vitessceErrors, setVitessceErrors] = useState([]);
-  const [isVisibleFirefoxWarning, setIsVisibleFirefoxWarning] = useState(sniffBrowser() === 'Firefox');
+  // Show a warning if the user is using Firefox.
+  useEffect(() => {
+    if (sniffBrowser() === 'Firefox' && !localStorage.getItem(localStorageFirefoxWarningKey)) {
+      toastError(FIREFOX_WARNING);
+      localStorage.setItem(localStorageFirefoxWarningKey, true);
+    }
+  }, [toastError]);
+
+  // Show instructions for exiting full screen mode when the user enters full screen mode.
+  useEffect(() => {
+    if (vizIsFullscreen) {
+      toastInfo('Press [esc] to exit full window.');
+    }
+  }, [vizIsFullscreen, toastInfo]);
```

Comment:
I think I prefer to pull related hooks together into a single hook, but it's entirely a preference. Do you prefer to have everything in a single file?

Reply:
It's on a case-by-case basis for me, but I lean towards keeping hooks focused on handling a single functionality/behavior:
- If there's an interdependent process where separating it leads to the output of one function being fed into another function, I do like to unify those into one on at least some level (e.g. if the SWR output from one fetch is used to set the params of a second fetch, it's helpful to have a single hook which abstracts that detail out from the component, but still keep the other two SWR calls as their own hooks)
- If it's two different conditions that trigger similar effects, I still keep them separated in order to keep the code maintainable and prevent unnecessary computations
  - RE: `maintainable` - as time goes on, combining more and more business logic into one hook/into one function leads to big fragile monoliths where 
  - RE: `unnecessary computations` - in this case, combining these two `useEffect`s into one would lead to extra `sniffBrowser` executions/potential extra `localStorage` reads whenever the visualization is full-screened; these are trivial operations, but in other circumstances these could be e.g. lookups from an API.